filecoin-proofs/src/bin/phase2.rs

```diff
@@ -355,492 +436,664 @@ fn create_initial_params<Tree: 'static + MerkleTreeTrait>(
-/// first parameters file in `param_paths` to the second file, then verifies that the second
-/// contribution hash transitions the second parameters file to the third file.
-fn verify_param_transitions(param_paths: &[&str], contribution_hashes: &[[u8; 64]]) {
-    let start_total = Instant::now();
+fn convert_small<'a>(path_before: &'a str) {
+    let (proof, hasher, sector_size, head, param_number, param_size, read_raw) =
+        parse_params_filename(path_before);
 
-    assert_eq!(
-        param_paths.len() - 1,
-        contribution_hashes.len(),
-        "the number of contributions must be one less than the number of parameter files"
+    let write_raw = !read_raw;
+
+    info!(
+        "converting params for circuit: {}-{}-{}-{}-{} {}->{}",
+        proof.pretty_print(),
+        hasher.pretty_print(),
+        sector_size.pretty_print(),
+        head,
+        param_size.pretty_print(),
+        param_number,
+        if write_raw { "raw" } else { "xxx" } // FIXME
     );
 
-    let mut next_params_before: Option<phase2::MPCParameters> = None;
+    // Default to small params for first participant.
+    let path_after = params_filename(
+        proof,
+        hasher,
+        sector_size,
+        &head,
+        param_number,
+        ParamSize::Small,
+        write_raw,
+    );
 
-    for (param_pair, provided_contribution_hash) in
-        param_paths.windows(2).zip(contribution_hashes.iter())
-    {
-        let path_before = param_pair[0];
-        let path_after = param_pair[1];
+    let start_total = Instant::now();
 
-        info!(
-            "verifying transition:\n\tparams: {} -> {}\n\tcontribution: {}",
-            path_before,
-            path_after,
-            hex::encode(&provided_contribution_hash[..])
-        );
+    info!("converting");
 
-        // If we are verifying the first contribution read both `path_before` and `path_after`
-        // files. For every subsequent verification, move the previous loop's "after" params to this
-        // loop's "before" params then read this loop's "after" params file. This will minimize the
-        // number of expensive parameter file reads.
-        let params_before = match next_params_before.take() {
-            Some(params_before) => params_before,
-            None => {
-                info!("reading 'before' params from disk: {}", path_before);
-                let file = File::open(path_before).unwrap();
-                let mut reader = BufReader::with_capacity(1024 * 1024, file);
-                let start = Instant::now();
-                let params_before = phase2::MPCParameters::read(&mut reader, true).unwrap();
-                info!(
-                    "successfully read 'before' params from disk, dt_read={}s",
-                    start.elapsed().as_secs()
-                );
-                params_before
-            }
-        };
+    info!(
+        "making streamer from small 'before' params: {}",
```

Comment:
The phase2-cli uses "before" to indicate params prior to a contribution is made, could you update this log to:

```rust
info!("creating streamer from non-raw params: {}", path_before);
```

Reply:
Done, but correctly indicating whether the source is raw or non-raw.